spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -636,17 +635,17 @@ public String or(String leftResult, String rightResult) {
         case NOT_NAN:
           return "not_nan(" + pred.ref().name() + ")";
         case LT:
-          return pred.ref().name() + " < " + sqlString(pred.literal());
+          return pred.ref().name() + " < " + pred.literal();
         case LT_EQ:
-          return pred.ref().name() + " <= " + sqlString(pred.literal());
+          return pred.ref().name() + " <= " + pred.literal();
         case GT:
-          return pred.ref().name() + " > " + sqlString(pred.literal());
+          return pred.ref().name() + " > " + pred.literal();
         case GT_EQ:
-          return pred.ref().name() + " >= " + sqlString(pred.literal());
+          return pred.ref().name() + " >= " + pred.literal();
         case EQ:
-          return pred.ref().name() + " = " + sqlString(pred.literal());
+          return pred.ref().name() + " = " + pred.literal();
         case NOT_EQ:
-          return pred.ref().name() + " != " + sqlString(pred.literal());
+          return pred.ref().name() + " != " + pred.literal();
```

Comment:
Why did we get rid of the `sqlString` function here? Is there a change that removes the need to quote strings for example?

Also, since the representation of `ByteBuffer` is potentially engine specific, would it be better to add the conversion to the `sqlString` here for Spark (e.g. using the leading capital X, like `X'123456'`).

This would allow other engines to handle it themselves (e.g. if Flink doesn't use leading `X` format).

I worry about unintended side effects from removing the explicit quoting as I've seen more PRs come up that handle adding and removing quotes as needed, but possibly you're more familiar with those conversations than I am.

Reply:
@kbendick is right. This should not remove the `sqlString` function. That converts a literal into a form that is usable by Spark. Instead, you should update that function to produce Spark's binary literal format.

Reply:
`sqlString` is implemented and used in the DescribeExpressionVisitor class.

I think this class is just for generating description information, and the description will be displayed in the log and Spark UI for debugging. So in my opinion, it is an acceptable way to use Literal.toString directly, which ensures that the description is the same as the style of other logs.

In fact, except for the difference between StringLiteral's single and double quotes, the current sqlString and Literal.toString are the same.

But if you think that keeping sqlString is a better way, I can also modify the code.
In addition, there are some differences between the description form of spark here, for example, the StringLiteral of spark does not have quotation marks. If you think that this should be the same as the behavior of spark, I can also modify the code.